File: java/StringCalculator.java
package com.someorganization.stringCalculator;

public class StringCalculator {

  public int add(String numbersWithDelimiter) throws IllegalArgumentException {
    if (numbersWithDelimiter.isEmpty()) {
      return 0;
    }

    String delimiter = this.getDelimiter(numbersWithDelimiter);

    String numberString = numbersWithDelimiter
        .replaceAll("//", "")
        .replaceAll("[\\[\\]]", "");
    if (!Character.isDigit(numberString.charAt(0))) {
      numberString = numberString.substring(1);
    }

    String[] numbers = numberString.split(delimiter);

    return getSum(numbers);
  }

  int getSum(String[] numberList) {
    int answer = 0;
    StringBuilder error = new StringBuilder();

    for (String s : numberList) {
      s = s.replace("\n", "")
          .replace("\r", "");
      int value = Integer.parseInt(s);
      if (value < 0) {
        error.append(s).append(" ");
      } else if (value < 1001) {
        answer += value;
      }
    }
    if (error.length() > 0) throw new IllegalArgumentException("Negatives not allowed: " + error.toString());
    return answer;
  }


  String getDelimiter(final String input) {
    String characterDelimiter = "//";
    String delimiter = "[,\n\r]";
    if (input.indexOf("//[") == 0) {
      var indexOfEnd = input.indexOf("]\n");
      delimiter = input.substring("//[".length(), indexOfEnd);
    } else if (input.indexOf("//") == 0) {
      delimiter = input.substring(2, 3);
    }

    return delimiter;
  }


}

File: csharp/StringCalculator/StringCalculator.cs
using System;
using System.Text;
using System.Text.RegularExpressions;

namespace StringCalculator
{
    
    
    public static class StringCalculator
    {
          public static int Add(string inputWithDelimiter )
          {
              switch (inputWithDelimiter)
              {
                  case null:
                      throw new ArgumentNullException(nameof(inputWithDelimiter));
                  case "":
                      { return 0;}
              }

                  var delimiter = GetDelimiter(inputWithDelimiter);
              
                  var numberString = Regex.Replace(@inputWithDelimiter, @"//", "");
                  numberString = Regex.Replace(numberString, "[\\[\\]]", "");
                  var a = numberString.ToCharArray();
                  var firstChar = numberString.ToCharArray()[0];
                  if (!char.IsDigit(firstChar) &&  firstChar != '-')
                  {
                      numberString = numberString.Substring(1);
                  }

                  var rx = new Regex(
                      delimiter); 
                  var numbers = rx.Split(numberString);

                  return GetSum(numbers);
          }
        
        
        public static int GetSum(string[] numberList)
        {
            var answer = 0;
            var error = new StringBuilder();

            foreach (var number in numberList)
            {
                var replace = Regex.Replace(number, "\n", "");
                replace = Regex.Replace(number, "\r", "");
                var value = int.Parse(replace);

                if (value < 0)
                {
                    error.Append($"{value} ");
                }
                else if (value < 1001)
                {
                    answer += value;
                    var a = 10;
                }
            }

            if (error.Length > 0) throw new ArgumentException("Negatives are not allowed: " + error);
            return answer;
        }

        public static string GetDelimiter(string input)
        {
            var delimiter = "[,\n\r]";
            if (input.IndexOf("//[", StringComparison.Ordinal) == 0)
            {
                var indexOfEnd = input.IndexOf("]\n");
                var index = "//[".Length;
                if (index >= 0 && input.Length > index)
                {
                    delimiter = input.Substring(index, indexOfEnd - index);
                }
            }
            else if (input.IndexOf("//") == 0)
            {
                delimiter = input.Substring(2, 1);
            }

            return delimiter;
        }
    }
}
